src/compiler/types.rs

```diff
@@ -1097,33 +1208,111 @@ impl Type {
             _ => panic!("Expected union or intersection type"),
         }
     }
+
+    pub fn as_resolved_type(&self) -> &dyn ResolvedTypeInterface {
+        match self {
+            Type::ObjectType(object_type) => {
+                if !object_type.is_resolved() {
+                    panic!("Not resolved")
+                }
+                object_type
+            }
+            _ => panic!("Expected resolved type"),
+        }
+    }
+
+    pub fn as_resolvable_type(&self) -> &dyn ResolvableTypeInterface {
+        match self {
+            Type::ObjectType(object_type) => object_type,
+            _ => panic!("Expected resolvable type"),
+        }
+    }
 }
 
 impl TypeInterface for Type {
     fn flags(&self) -> TypeFlags {
         match self {
             Type::IntrinsicType(intrinsic_type) => intrinsic_type.flags(),
             Type::LiteralType(literal_type) => literal_type.flags(),
+            Type::ObjectType(object_type) => object_type.flags(),
             Type::UnionOrIntersectionType(union_or_intersection_type) => {
                 union_or_intersection_type.flags()
             }
         }
     }
+
+    fn maybe_symbol(&self) -> Option<Rc<Symbol>> {
```

Comment:
It's probably time to implement a similar boilerplate-generation macro for type types as I did for node types